clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -234,9 +229,10 @@
  * successful writes are marked as aborted, hence keeping the transactional guarantees.
  * </p>
  * <p>
- * This client can communicate with brokers that are version 0.10.0 or newer. Older or newer brokers may not support
- * certain client features.  For instance, the transactional APIs need broker versions 0.11.0 or later. You will receive an
- * <code>UnsupportedVersionException</code> when invoking an API that is not available in the running broker version.
+ * This client can communicate with brokers that are version 2.1 or newer. Older brokers may not support
```

Comment:
I removed the "or newer" from here and the consumer documentation because I couldn't think of any example of where that is true. Am I missing something @dajac @junrao ?

Reply:
For producers, transaction.version needs to be set on the newer server to enable txn V2 support.

For consumers, group.version needs to be set to enable the new consumer rebalance protocol.

Reply:
That makes sense. But I think the message as written before made it sound like a newer broker version may result in some consumer/producer features not working. That seems unlikely to me. The more likely thing is that a newer broker would mean a certain consumer version is no longer supported.